.github/actions/is-release-week-v1/src/run.test.ts

```diff
@@ -0,0 +1,98 @@
+import sinon, { type SinonFakeTimers } from 'sinon'
+import { assert } from 'chai'
+import run, { type Core } from './run'
+
+describe('run', () => {
+  afterEach(() => {
+    sinon.restore()
+  })
+
+  it('fails if oddWeek is not given', () => {
+    const core = {
+      getInput: sinon.stub(),
+      setFailed: sinon.spy()
+    }
+
+    run(core as unknown as Core)
+    assert.isTrue(
+      core.setFailed.calledOnceWith('`oddWeek` must be "true" or "false"')
+    )
+  })
```

Comment:
Oh interesting, so the `action.yml` adds a default value for `oddWeek` to be `false` but because we're testing it directly it doesn't get passed here.

(see comment below) if no input is given it should through the action before it reaches your is true or false statement check. I would've thought the `catch` would get the thrown error from the `{ required: true}` in your `getInput`

Reply:
The test stubs out `getInput` so it never called into the original `getInput` code to generate the throw.

Reply:
If we want to test that we have the `required: true` added to the getInput call we could ensure that it was called with it:

```js
// don't know for certain if this is works as written, but should get the point across
assert.isTrue(core.getInput.calledWith('oddWeek, sinon.match({ required: true }));
```